tlp-pd.in

```diff
@@ -0,0 +1,394 @@
+        # Callback for tlp after profile changes: update internal state only
+        if profile not in AVAILABLE_PROFILES:
+            raise dbus.exceptions.DBusException(
+                f"Invalid profile: {profile}",
+                name="org.freedesktop.DBus.Error.InvalidArgs",
+            )
+
+        if args.debug:
+            syslog.syslog(syslog.LOG_DEBUG, f"SyncProfile: {profile}")
+
+        # Update active profile
+        self._active_profile = profile
+
+        # Notify the desktop about the new profile using the signal
+        self.PropertiesChanged(INTERFACE_NAME, {"ActiveProfile": profile}, [])
+
+    @dbus.service.signal(dbus_interface=INTERFACE_NAME, signature="u")
+    # Exported on the interface org.freedesktop.UPower.PowerProfiles
+    # Emitted when a profile hold is released due to manual profile change
+    def ProfileReleased(self, cookie):
+        if args.debug:
+            syslog.syslog(syslog.LOG_DEBUG, f"Signal ProfileReleased")
+        pass
+
+    @dbus.service.signal(dbus_interface=dbus.PROPERTIES_IFACE, signature="sa{sv}as")
+    # Exported on the interface org.freedesktop.DBus.Properties
+    # Emitted when the active profile changes
+    def PropertiesChanged(
+        self, interface_name, changed_properties, invalidated_properties
+    ):
+        if args.debug:
+            syslog.syslog(
+                syslog.LOG_DEBUG,
+                f"Signal PropertiesChanged: interface={interface_name}, changed_properties={changed_properties}, invalidated_properties={invalidated_properties}",
+            )
+        pass
+
+    # --- Interface org.freedesktop.DBus.Properties
+
+    @dbus.service.method(
+        dbus_interface=dbus.PROPERTIES_IFACE, in_signature="ss", out_signature="v"
+    )
+    def Get(self, interface_name, property_name):
+        # Get a property value
+        if args.debug:
+            syslog.syslog(
+                syslog.LOG_DEBUG,
+                "Method Get: interface={interface_name}, property={property_name}",
+            )
+
+        if interface_name not in [INTERFACE_NAME, INTERFACE_NAME_LEGACY]:
+            raise dbus.exceptions.DBusException(
+                f"Unknown interface: {interface_name}",
+                name="org.freedesktop.DBus.Error.UnknownInterface",
+            )
+
+        return self._get_property(property_name)
+
+    @dbus.service.method(
+        dbus_interface=dbus.PROPERTIES_IFACE, in_signature="s", out_signature="a{sv}"
+    )
+    def GetAll(self, interface_name):
+        # Get all properties
+        if args.debug:
+            syslog.syslog(
+                syslog.LOG_DEBUG, f"Method GetAll: interface={interface_name}"
+            )
+
+        if interface_name not in [INTERFACE_NAME, INTERFACE_NAME_LEGACY]:
+            raise dbus.exceptions.DBusException(
+                f"Unknown interface: {interface_name}",
+                name="org.freedesktop.DBus.Error.UnknownInterface",
+            )
+
+        return {
```

Comment:
I'd suggest making this a series of self._get_property calls, so if the implementation of a property changes that can be updated in one place. More generally, I really liked how other dbus libraries let you write properties as python properties instead of intertwined with the implementations of Get and Set, unfortunately this doesn't come for free with python-dbus but it's probably not too hard to write a custom bit of abstraction for this project.